wpilibc/src/main/native/cpp/DoubleSolenoid.cpp

```diff
@@ -49,6 +49,12 @@ DoubleSolenoid::DoubleSolenoid(std::shared_ptr<PneumaticsBase> module,
   m_reverseMask = 1 << reverseChannel;
   m_mask = m_forwardMask | m_reverseMask;
 
+  if (m_module->CheckAndReserveSolenoids(m_mask) != 0) {
+    // TODO tell which solenoid is already allocated
```

Comment:
Can you just check and reserve forwardMask separate from reserveMask to address the TODO?